src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -111,14 +112,14 @@ public DirectoryController(DirectoryService service) {
         return ResponseEntity.ok().body(service.deleteElement(elementUuid, userId));
     }
 
-    @RequestMapping(method = RequestMethod.HEAD, value = "/directories/{directoryUuid}/{elementName}")
-    @Operation(summary = "Check if the element exists")
+    @RequestMapping(method = RequestMethod.HEAD, value = "/directories/{directoryUuid}/{elementName}/{type}")
+    @Operation(summary = "Check if the element exists in the given directory")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The element exists"),
                            @ApiResponse(responseCode = "404", description = "The element doesn't exist")})
     public ResponseEntity<Mono<Void>> elementExists(@PathVariable("directoryUuid") UUID directoryUuid,
                                                     @PathVariable("elementName") String elementName,
-                                                    @RequestHeader("userId") String userId) {
-        return ResponseEntity.ok().body(service.elementExists(directoryUuid, elementName, userId));
+                                                    @PathVariable("type") String type) {
+        return ResponseEntity.ok().body(service.elementExistsMono(directoryUuid, elementName, type));
```

Comment:
HEAD request method shouldn't return a body ! It was like that before but it's not good.
cf. warning : https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD